From b3ced4353a48dfc657d28f605ae10850f971bca0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 22 Aug 2026 03:03:31 +0000
Subject: [PATCH 25/36] tests: materialize QStringBuilder chains fed into
 QTest::newRow() <<

QTestData::operator<<() is a template that deduces its metatype from
the argument, then registers it via qMetaTypeId<T>() -- if the
argument is a raw, unmaterialized QStringBuilder chain instead of a
QString, T deduces to the builder's own type, which was never meant
to be (and isn't) a registered metatype: same underlying problem as
QMetaTypeId<T>::qt_metatype_id() missing everywhere else in this
series, just reached via QTest's data-table mechanism instead of
QVariant. Wrapped each raw concatenation in QString(...) before it
hits the << operator, in test_Composer_responses.cpp (1 row) and
test_Html_formatting.cpp (1 row, 3 columns).

Swept the rest of the test suite for the same "QTest::newRow(...) <<
<unwrapped concatenation>" shape -- no other instances found.
---
 tests/Composer/test_Composer_responses.cpp | 4 ++--
 tests/Composer/test_Html_formatting.cpp    | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/tests/Composer/test_Composer_responses.cpp b/tests/Composer/test_Composer_responses.cpp
index 5f4d9005..1ee00e86 100644
--- a/tests/Composer/test_Composer_responses.cpp
+++ b/tests/Composer/test_Composer_responses.cpp
@@ -554,8 +554,8 @@ void ComposerResponsesTest::testFormatFlowedComposition_data()
 
     QString longQuotePrefix = QString(76, QLatin1Char('>')) + QLatin1Char(' ');
     QTest::newRow("long-quote-prefix")
-            << longQuotePrefix + QLatin1String("hello world!")
-            << longQuotePrefix + QLatin1String("hello world!");
+            << QString(longQuotePrefix + QLatin1String("hello world!"))
+            << QString(longQuotePrefix + QLatin1String("hello world!"));
 
     // In the next test, the best approach would be wrapping the "QtObject" into the next line (and perhaps merging it
     // with the contents in there). However, doing so would require insight into whether that line was originally flowed
diff --git a/tests/Composer/test_Html_formatting.cpp b/tests/Composer/test_Html_formatting.cpp
index 2847b7cc..dd281543 100644
--- a/tests/Composer/test_Html_formatting.cpp
+++ b/tests/Composer/test_Html_formatting.cpp
@@ -402,10 +402,10 @@ void HtmlFormattingTest::testPlainTextFormattingViaPaste_data()
     QString shortLipsum = QString(lipsum + QLatin1Char(' ') + lipsum).left(5*160);
 
     QTest::newRow("collapsed-last-quote")
-            << QStringLiteral("Some real text.\n> ") + lipsum + QLatin1Char(' ') + lipsum
-            << QStringLiteral("Some real text.\n> ") + shortLipsum + " ...\n"
+            << QString(QStringLiteral("Some real text.\n> ") + lipsum + QLatin1Char(' ') + lipsum)
+            << QString(QStringLiteral("Some real text.\n> ") + shortLipsum + " ...\n")
             << QString()
-            << QStringLiteral("Some real text.\n> ") + lipsum + QLatin1Char(' ') + lipsum + "\n";
+            << QString(QStringLiteral("Some real text.\n> ") + lipsum + QLatin1Char(' ') + lipsum + "\n");
 
     QTest::newRow("quoted-common")
             << QStringLiteral("On quinta-feira, 4 de outubro de 2012 15.46.57, André Somers wrote:\n"
-- 
2.43.0

